query-input/src/main/java/de/brandwatch/minianalytics/queryinput/controller/Controller.java

```diff
@@ -0,0 +1,21 @@
+package de.brandwatch.minianalytics.queryinput.controller;
+
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@org.springframework.stereotype.Controller
```

Comment:
That's a bit unusual. Why not import it and just use `@Controller`?

Reply:
Because the class is also named Controller. So the file wont compile with `@Controller` annotation

Reply:
Right. I would generally advice against that kind of naming altogether, i.e. as a default don't name things like library classes or annotations. Not only because the imports, but also because those names are usually very generic, and too generic for your purpose. i.e. the class name `Controller` doesn't tell me anything about what it's doing. There can certainly be multiple classes that have the `@Controller` annotation, but do different things. 

This seems to serve the web interface, so maybe call it `WebController` or `InterfaceController` or something like that. The other one could be the `QueryController`

Reply:
Ok, I renamed them.
But my problem is more general. I often don't know how to name things like the `Controller` or `RestController`. Is there any specific schema you name your classes?

Reply:
Not beyond I try to give things a sensible name the tells me what it is about. 

On the book shelf in the ping pong room there is a book called "Clean Code" that has a whole chapter called "Meaningful Names". Feel free to check it out ;) (also, there's other great books there ...)

Reply:
Definitly will have a look at it